pyproject_metadata/__init__.py

```diff
@@ -559,19 +559,26 @@ def validate(self, *, warn: bool = True) -> None:
             msg = 'Setting "project.license" to an SPDX license expression is not compatible with "License ::" classifiers'
             raise ConfigurationError(msg)
 
-        if warn and self.auto_metadata_version not in PRE_SPDX_METADATA_VERSIONS:
-            if isinstance(self.license, License):
+        if warn:
+            if self.description and '\n' in self.description:
                 warnings.warn(
-                    'Set "project.license" to an SPDX license expression for metadata >= 2.4',
-                    ConfigurationWarning,
-                    stacklevel=2,
-                )
-            elif any(c.startswith('License ::') for c in self.classifiers):
-                warnings.warn(
-                    '"License ::" classifiers are deprecated for metadata >= 2.4, use a SPDX license expression for "project.license" instead',
+                    'The one-line summary "project.description" should not contain more than one line. Backends will truncate or merge lines.',
```

Comment:
What is a backend in this context? I think that all users of pyproject-metadata do not patch the metadata to truncate or merge lines. I think that what happens and what package authors expect is for lines to be concatenated. Anyhow, I really think that the "one line" in the specification refers to the length of the description, not to the fact that it should technically not contain line return characters. I think that concatenating the lines is the thing to do. Therefore I would do so and make the warning say just that.

Reply:
Some backends (that do not use pyproject-metadata) will truncate on the first line - specifically, setuptools does this. If they are using pyproject-metadata, though, we are merging them, so the error message is slightly incorrect.

```suggestion
                    'The one-line summary "project.description" should not contain more than one line. Merging; some build-backends may truncate instead.',
```

?